test/EdgeFactory/EdgeFactory.t.sol

```diff
@@ -0,0 +1,313 @@
+        adapterParams[0] = IEdgeFactory.AdapterParams(address(assetTST), unitOfAccount, address(stubOracle));
+        adapterParams[1] = IEdgeFactory.AdapterParams(address(assetTST2), unitOfAccount, address(stubOracle));
+
+        address[] memory externalVaults = new address[](0);
+        IEdgeFactory.RouterParams memory routerParams = IEdgeFactory.RouterParams(externalVaults, adapterParams);
+
+        IEdgeFactory.LTVParams[] memory ltvParams = new IEdgeFactory.LTVParams[](1);
+        ltvParams[0] = IEdgeFactory.LTVParams(0, 1, 0.8e4, 0.9e4);
+
+        IEdgeFactory.DeployParams memory deployParams =
+            IEdgeFactory.DeployParams(vaultParams, routerParams, ltvParams, unitOfAccount);
+
+        stubOracle.setPrice(address(assetTST), address(unitOfAccount), 1e18);
+        stubOracle.setPrice(address(assetTST2), address(unitOfAccount), 1e18);
+
+        vm.expectRevert();
+        edgeFactory.deploy(deployParams);
+    }
+
+    function testDeploy_HandlesExternalVaults() public {
+        IEdgeFactory.VaultParams[] memory vaultParams = new IEdgeFactory.VaultParams[](2);
+        vaultParams[0] = IEdgeFactory.VaultParams(address(assetTST), irm, false);
+        vaultParams[1] = IEdgeFactory.VaultParams(address(assetTST2), irm, true);
+
+        IEdgeFactory.AdapterParams[] memory adapterParams = new IEdgeFactory.AdapterParams[](2);
+        adapterParams[0] = IEdgeFactory.AdapterParams(address(assetTST), unitOfAccount, address(stubOracle));
+        adapterParams[1] = IEdgeFactory.AdapterParams(address(assetTST2), unitOfAccount, address(stubOracle));
+
+        address[] memory externalVaults = new address[](2);
+        externalVaults[0] = address(assetTST);
+        externalVaults[1] = address(assetTST2);
+        IEdgeFactory.RouterParams memory routerParams = IEdgeFactory.RouterParams(externalVaults, adapterParams);
+
+        IEdgeFactory.LTVParams[] memory ltvParams = new IEdgeFactory.LTVParams[](1);
+        ltvParams[0] = IEdgeFactory.LTVParams(1, 0, 0.8e4, 0.9e4);
+
+        IEdgeFactory.DeployParams memory deployParams =
+            IEdgeFactory.DeployParams(vaultParams, routerParams, ltvParams, unitOfAccount);
+
+        vm.mockCall(
+            address(assetTST), abi.encodeWithSelector(IERC4626.asset.selector), abi.encode(makeAddr("assetTST.asset"))
+        );
+        vm.mockCall(
+            address(assetTST2), abi.encodeWithSelector(IERC4626.asset.selector), abi.encode(makeAddr("assetTST2.asset"))
+        );
+        stubOracle.setPrice(address(assetTST), address(unitOfAccount), 1e18);
+        stubOracle.setPrice(address(assetTST2), address(unitOfAccount), 1e18);
+
+        (address router, address[] memory vaults) = edgeFactory.deploy(deployParams);
+        _verifyDeployment(router, vaults, deployParams);
+    }
+
+    function _verifyDeployment(address router, address[] memory vaults, IEdgeFactory.DeployParams memory deployParams)
```

Comment:
if you like, you can make a use of `PerspectiveVerifier` here: https://github.com/euler-xyz/evk-periphery/blob/bfbdc08b3a08b3919c1e9d1e53aaeb2b88396157/script/utils/PerspectiveCheck.s.sol#L18

you can call it i.e. as follows: https://github.com/euler-xyz/evk-periphery/blob/bfbdc08b3a08b3919c1e9d1e53aaeb2b88396157/script/production/base/clusters/BaseCluster.s.sol#L182

the idea is that you can verify given vault against any perspective even if it's not fully conformant. you can specify which errors you expect and which should be ignored. in your case, you should pass the ungoverned 0x perspective. if you don't want to simulate whitelisting oracle adapters and IRMs, you can simply expect those errors